WordPress/gutenberg
# Build: using a postcss plugin to generate the admin-schemes styles

This PR adds a PostCSS plugin to generate the admin-schemes automatically at build-time.
This has some pros over the previous approach.

 - We don't need to write the specific rules ourselves, they are automatically extracted from the styles
 - The admin schemes styles are generated per stylesheet which means they are not generated in the `edit-post` style. If you load just the components stylesheet, it will come bundled with the admin-schemes styles.

**Usage**:

Instead of using colors directly in our rules, we need to use these helpers:

 `color: theme(primary)`

The argument passed to the `theme` helper is one of the defined theme colors (primary/secondary/primary-dark15 for now, we can add as much as colors as we want)

**Some drawbacks:**

 - We can't use `darken/lighten` on theme colors, because the Sass loader is run before the post-css transformation, for example I had to define a `primary-dark15` theme color to replace the previous call to `darken( theme(primary), 15 )`

 - The date picker styles were overriden using sass variables and since the date picker styles uses darken/lighten, we can't do this anymore, we'd have to figure out another way to override those.

This PR will help with #6562 

**Testing instructions**

 - Switch your user's profile and check that the theme colors are updating properly.
